sdks/java/core/src/test/java/org/apache/beam/sdk/coders/CommonCoderTest.java

```diff
@@ -0,0 +1,194 @@
+package org.apache.beam.sdk.coders;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.CharStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.beam.sdk.coders.Coder.Context;
+import org.apache.beam.sdk.util.CoderUtils;
+import org.apache.beam.sdk.values.KV;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests that Java SDK coders standardized by the Fn API meet the common spec.
+ */
+@RunWith(Parameterized.class)
+public class CommonCoderTest {
+  private static final String STANDARD_CODERS_YAML_PATH =
+      "/org/apache/beam/fn/v1/standard_coders.yaml";
+
+  private static final Map<String, Class<?>> coders = ImmutableMap.<String, Class<?>>builder()
```

Comment:
I agree with your sentiment that in the absence of a core model in the repo - which in [#662](https://github.com/apache/beam/pull/662) I dropped into `mode/pipeline` - it seems to make sense to put this in a Fn API bucket.

But I have two substitutions that I think will actually matter:

1. `sdks` -> `model` since the model is going to be part of both SDKs and runners. It isn't so bad to put under `sdks` but `model` is a better place for protos and generic test data. I could be convinced otherwise.
2. `fn_api` -> `core` or `pipeline` or `coders` or some such, since this is a restriction placed on things that don't have a direct dependency on the Fn API, or rather a restriction that the Fn API and the core SDKs share, making sure that we don't introduce a reversed dependency arrow from the harness to the core. Misc thought: the Fn API is likely to evolve more frequently than other bits of the model.